main.tf

```diff
@@ -194,20 +194,20 @@ resource "aws_instance" "lx" {
       ${element(data.template_file.lx_script_preface.*.rendered, count.index)}
       ${join("", data.template_file.lx_test.*.rendered)}
       HEREDOC
-    destination = "~/watchmaker_test.sh"
+    destination = "~/inline-${local.ami_underlying[element(local.lx_requests, count.index)]}.sh"
   }
 
   provisioner "remote-exec" {
     inline = [
-      "chmod +x ~/watchmaker_test.sh",
-      "~/watchmaker_test.sh",
+      "chmod +x ~/inline-${local.ami_underlying[element(local.lx_requests, count.index)]}.sh",
+      "~/inline-${local.ami_underlying[element(local.lx_requests, count.index)]}.sh",
     ]
 
     connection {
       host = coalesce(self.public_ip, self.private_ip)
       type = "ssh"
       # this is where terraform puts the above inline script
-      script_path = "~/inline.sh"
+      script_path = "~/inline-mini-${local.ami_underlying[element(local.lx_requests, count.index)]}.sh"
```

Comment:
bam! can we do that also for the winrm connection on windows?

Reply:
This should work...